packages/frontend/src/components/common/ResourceView/ResourceIcon.tsx

```diff
@@ -141,4 +142,74 @@ const ResourceTypeIcon: FC<ResourceTypeIconProps> = ({ type }) => {
     }
 };
 
-export { ResourceIcon, ResourceTypeIcon };
+export const ResourceIconWithIndicator: FC<{
+    children: ReactNode;
+    disabled: boolean;
+    tooltipLabel: ReactNode;
+}> = ({ disabled, tooltipLabel, children }) => {
+    // NOTE: Control the Tooltip visibility manually to allow hovering on Label.
+    const [opened, setOpened] = useState(false);
+    const [isHovering, setIsHovering] = useState(false);
+    const closeTimeoutId = useRef<number | undefined>(undefined);
+
+    const handleMouseEnter = () => {
+        clearTimeout(closeTimeoutId.current);
+        setOpened(true);
+    };
+
+    const handleMouseLeave = () => {
+        // NOTE: Provide similar delay as Tooltip component
+        closeTimeoutId.current = window.setTimeout(() => {
+            setOpened(false);
+        }, 100);
+    };
+
+    const handleLabelMouseEnter = () => {
+        setIsHovering(true);
+        clearTimeout(closeTimeoutId.current);
+    };
+
+    const handleLabelMouseLeave = () => {
+        setIsHovering(false);
+        // NOTE: Provide similar delay as Tooltip component
+        closeTimeoutId.current = window.setTimeout(() => {
+            setOpened(false);
+        }, 100);
+    };
+
+    return (
+        <Indicator
+            disabled={disabled}
+            position="bottom-end"
+            color="transparent"
+            label={
+                <Tooltip
+                    w={300}
+                    withinPortal
+                    multiline
+                    offset={-2}
+                    position="bottom"
+                    sx={{ pointerEvents: 'auto' }}
+                    label={
+                        <div
+                            onMouseEnter={handleLabelMouseEnter}
+                            onMouseLeave={handleLabelMouseLeave}
+                        >
+                            {tooltipLabel}
+                        </div>
+                    }
+                    opened={opened || isHovering}
+                >
+                    <MantineIcon
+                        fill="red"
+                        icon={IconAlertTriangle}
```

Comment:
perhaps we want this to be a prop, rather than being tied to a specific color, what do we think?